ui/app/AppLayouts/Chat/panels/UserListPanel.qml

```diff
@@ -75,66 +76,128 @@ Item {
 
         clip: true
 
-        StatusListView {
-            id: userListView
-            objectName: "userListPanel"
-
-            clip: false
+        Loader {
+            id: loadingUsersView
 
             anchors.fill: parent
-            anchors.bottomMargin: Style.current.bigPadding
-            displayMarginEnd: anchors.bottomMargin
 
-            model: SortFilterProxyModel {
-                sourceModel: root.usersModel
+            active: messageStore.loading
+            visible: active
+            sourceComponent: MessagesLoadingView {
+                // anchors.margins: 16
+                isUserList: true
+                anchors.fill: parent
+            }
+        }
+
+
+        Repeater {
+            id: chatRepeater
+            model: chatCommunitySectionModule && chatCommunitySectionModule.model
+
+            Loader {
+                id: listLoader
 
-                proxyRoles: FastExpressionRole {
-                    function displayNameProxy(nickname, ensName, displayName, aliasName) {
-                        return ProfileUtils.displayName(nickname, ensName, displayName, aliasName)
+                property bool makeActive: model.type !== Constants.chatType.category && model.type !== Constants.chatType.unknown && model.loaderActive
+                property string chatId: model.itemId
+
+                anchors.fill: parent
+                active: false
+                asynchronous: true
+
+                onMakeActiveChanged: {
+                    if (!listLoader.active && listLoader.makeActive) {
+                        loadingUsersView.active = true
+                        userListDelay.start()
+                    } else if (!listLoader.makeActive) {
+                        listLoader.active = false
                     }
-                    name: "preferredDisplayName"
-                    expectedRoles: ["localNickname", "ensName", "displayName", "alias"]
-                    expression: displayNameProxy(model.localNickname, model.ensName, model.displayName, model.alias)
                 }
 
-                sorters: [
-                    RoleSorter {
-                        roleName: "onlineStatus"
-                        sortOrder: Qt.DescendingOrder
-                    },
-                    StringSorter {
-                        roleName: "preferredDisplayName"
-                        caseSensitivity: Qt.CaseInsensitive
+                Timer {
+                    id: userListDelay
+                    repeat: false
+                    running: false
+                    interval: 500
```

Comment:
Is it not causing issues if switching between channels there and back faster than in 500ms?

Reply:
Let me try. I doubt it would though, since the Timer is per MemberList

Reply:
It looks fine. Ofc, switching fast makes it lag a little because of so many loads at once, but once it's loaded, everything is smooth.

[fast-switch.webm](https://github.com/user-attachments/assets/d97dfbf1-8944-42f8-ac61-cc543f27e1aa)